modules/tasks/tasks/src/test/java/org/motechproject/tasks/it/ActionParametersIT.java

```diff
@@ -0,0 +1,228 @@
+import org.motechproject.tasks.service.util.TaskContext;
+import org.motechproject.testing.osgi.BasePaxIT;
+import org.motechproject.testing.osgi.container.MotechNativeTestContainerFactory;
+import org.motechproject.testing.osgi.wait.Wait;
+import org.motechproject.testing.osgi.wait.WaitCondition;
+import org.ops4j.pax.exam.ExamFactory;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerSuite;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.junit.Assert.assertTrue;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerSuite.class)
+@ExamFactory(MotechNativeTestContainerFactory.class)
+public class ActionParametersIT extends BasePaxIT {
+
+    private static final String TASK_TEST_CHANNEL_NAME = "motech-tasks-test-bundle";
+    private static final String TASK_TEST_SERVICE_INTERFACE = "org.motechproject.testmodule.service.TasksTestService";
+    private static final String TASK_TEST_SERVICE_METHOD = "createTestObjectWithPostActionParameter";
+    private static final String TASK_TEST_ACTION_NAME = "Test Action return Post Action Params";
+    private static final String MDS_CHANNEL_NAME = "org.motechproject.motech-platform-dataservices-entities";
+    private static final String VERSION = "0.29";
+    private static final String TRIGGER_SUBJECT = "mds.crud.serverconfig.SettingsRecord.CREATE";
+
+    private static final Integer MAX_RETRIES_BEFORE_FAIL = 20;
+    private static final Integer WAIT_TIME = 2000;
+
+    @Inject
+    private ChannelService channelService;
+
+    @Inject
+    private TaskService taskService;
+
+    @Inject
+    private TriggerHandler triggerHandler;
+
+    @Inject
+    private SettingsDataService settingsDataService;
+
+    @Inject
+    private TaskActivitiesDataService taskActivitiesDataService;
+
+
+    private MotechJsonReader motechJsonReader = new MotechJsonReader();
+
+    @Override
+    protected Collection<String> getAdditionalTestDependencies() {
+        return Arrays.asList(
+                "org.motechproject:motech-tasks-test-bundle"
+        );
+    }
+
+    @Before
+    public void setUp() throws IOException, InterruptedException {
+        createAdminUser();
+        login();
+        Channel channel = loadChannel();
+        setUpSecurityContext("motech", "motech", "manageTasks");
+
+        channelService.addOrUpdate(channel);
+        waitForChannel(MDS_CHANNEL_NAME);
+    }
+
+    @Test
+    public void testActionWithPostActionParameters() throws InterruptedException, IOException, ActionNotFoundException {
+        long taskID = createTestTask();
```

Comment:
do not check PostActionParameterObject, it is implementation detail, you should check if TaskTestObject has proper field value after action executed